src/config.rs

```diff
@@ -140,25 +135,92 @@ impl fmt::Display for ConfigurationError {
     }
 }
 
-// Loads the configuration file from command arguments or defaults to config.toml
+// Loads the configuration file from command arguments and the environment.
 pub fn load_configuration() -> Result<Config, ConfigurationError> {
     //load arguments
-    let args: Vec<String> = env::args().collect();
-    // We default to loading a config.toml if minsql is run without arguments
-    let mut filename = "config.toml";
-    if args.len() > 1 {
-        filename = args[1].as_str();
-    }
-    // try to read the file
-    let contents = match fs::read_to_string(filename) {
-        Ok(f) => f,
-        Err(_) => return Err(ConfigurationError::new("Could not read configuration file")),
+    let matches = App::new("MinSQL")
+        .version("1.0")
+        .about("Log Search Engine")
+        .arg(
+            Arg::with_name("address")
+                .takes_value(true)
+                .default_value(DEFAULT_SERVER_ADDRESS)
+                .short("a")
+                .long("address")
+                .help("Server binding address, i.e.: 0.0.0.0:9000")
+                .required(true),
+        )
+        .get_matches();
+
+    // Server address, safe to unwrap since it has a default value.
+    let address = matches.value_of("address").unwrap().to_string();
+
+    // Check for configuration on the environment, else return error.
+
+    let metadata_endpoint: String = match env::var("METABUCKET_ENDPOINT") {
+        Ok(val) => val,
+        Err(e) => {
+            return Err(ConfigurationError::new(&format!(
+                "No meta bucket endpoint environment variable `METABUCKET_ENDPOINT` set. {}",
+                e
+            )));
+        }
+    };
+
+    let metadata_bucket: String = match env::var("METABUCKET_NAME") {
+        Ok(val) => val,
+        Err(e) => {
+            return Err(ConfigurationError::new(&format!(
+                "No meta bucket name environment variable `METABUCKET_NAME` set. {}",
+                e
+            )));
+        }
     };
-    // try to parse the toml string
-    let mut configuration: Config = match toml::from_str(&contents) {
-        Ok(t) => t,
-        Err(e) => return Err(ConfigurationError::new(&format!("{}", e)[..])),
+
+    let access_key: String = match env::var("METABUCKET_ACCESS_KEY") {
```

Comment:
Use a constant for env var names as a good practice: https://doc.rust-lang.org/rust-by-example/custom_types/constants.html